src/stdlib_string_type.fypp

```diff
@@ -421,7 +421,11 @@ contains
         type(string_type), intent(in) :: string
         integer :: ich
 
-        ich = merge(ichar(string%raw), 0, allocated(string%raw))
+        if (allocated(string%raw) .and. len(string) > 0) then
+          ich = ichar(string%raw(1:1))
+        else
+          ich = 0
+        end if
```

Comment:
May I ask what was the issue with the original code?

Reply:
Accordings to the book  '_Modern Fortran explained (Incorporating Fortran 2018)'_,  argument  of '_ichar_' is a character which length equal one. 
'string%raw' can be more than length one. In such case, runtime error occurred when use nagfor. On the other hands, gfortran seems to accept string argument.

Reply:
Thank you @sakamoti for your answer. But I was wondering why the `merge` option was removed.

Reply:
`merge` doesn't necessarily short-circuit the evaluation of its arguments like a ternary operator in C would. This could lead to an access of the uninitialized raw payload of the `string_type` which is an error.

Reply:
Thanks @awvwgk for the explanation.

Reply:
I havn't know  which one is more efficiently to select 'merge' or 'if construct'. I just felt a bit complex to write with 'merge'. And I want to acces 'string%raw' only when 'conditions' becoms true. So, I select to write with 'if construct'.
Thank you @awvwgk for the explanation.